discretize/base/base_mesh.py

```diff
@@ -1880,6 +1880,46 @@ def get_edge_inner_product(
             f"get_edge_inner_product not implemented for {type(self)}"
         )
 
+
+    def get_edge_inner_product_surface(
+        self,
+        model=None,
+        invert_model=False,
+        invert_matrix=False,
+        do_fast=True,
+        **kwargs,
+    ):
+
+        raise NotImplementedError(
+            f"get_edge_inner_product_surface not implemented for {type(self)}"
+        )
+
+    def get_face_inner_product_surface(
+        self,
+        model=None,
+        invert_model=False,
+        invert_matrix=False,
+        do_fast=True,
+        **kwargs,
+    ):
+
+        raise NotImplementedError(
```

Comment:
You could actually implement this one for all mesh types at the same time, As the operation you need to do is just:

```
diag(face_areas * model)
```

and `face_areas` is defined for all mesh types.

Reply:
done